### PR TITLE
Provide Arc configuration allowing to fail when interceptors used on private methods

### PR DESCRIPTION
**Pull request by geoand**
Follow up of: #21260

### CODE REVIEW

File: extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcConfig.java [hunk 1]
```diff
@@ -69,6 +69,15 @@
     @ConfigItem(defaultValue = "true")
     public boolean transformUnproxyableClasses;
 
+    /**
+     * If set to true, the build will fail if an annotation that would result in an interceptor being created (such as
```
Comment (mkouba) [inline extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcConfig.java:73]: Formally, the build should fail if a private method that is neither an observer nor a producer and is annotated with an interceptor binding. The transactional example is OK and should follow ;-).

File: independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Methods.java [hunk 2]
```diff
@@ -255,14 +256,21 @@ static boolean isObjectToString(MethodInfo method) {
             if (Modifier.isPrivate(method.flags())
                     && !Annotations.contains(methodAnnotations, DotNames.OBSERVES)
                     && !Annotations.contains(methodAnnotations, DotNames.OBSERVES_ASYNC)) {
```
Comment (mkouba) [inline independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Methods.java:258]: I think we should also add a check for `@Produces`...

File: extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcConfig.java [hunk 3]
```diff
@@ -69,6 +69,15 @@
...
+     * {@code @Transactional})
+     * is placed on a private method.
+     * If set to false, Quarkus simply logs a warning that the annotation will be ignored.
+     */
+    @ConfigItem(defaultValue = "false")
+    public boolean failOnPrivateInterceptors;
```
Comment (mkouba) [inline extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcConfig.java:79]: It's just a nitpicking but we don't fail _"if a private interceptor is found"_ but if a _"private method is intercepted"_. So maybe something like `failOnInterceptedPrivateMethod`?

Comment (mkouba) [reply extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcConfig.java:79]: And the variables in other classes should be renamed as well.

Comment (mkouba) [reply extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcConfig.java:79]: @manovotn WDYT?

File: independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Methods.java (hunk 2 above)
Comment (geoand) [reply independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Methods.java:258]: Done

File: extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcConfig.java (hunk 3 above)
Comment (geoand) [reply extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcConfig.java]: I used your suggestion

Comment (manovotn) [reply extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcConfig.java]: +1, thanks for changing it

**Review by manovotn — APPROVED** [review]
Looks good

**Comment by mkouba** [comment]
@geoand Could you pls update the javadoc of `ArcConfig.failOnInterceptedPrivateMethod` as well? Otherwise looks very good.

**Comment by geoand** [comment]
@mkouba  done

File: extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcConfig.java [hunk 4]
```diff
@@ -69,6 +69,15 @@
     @ConfigItem(defaultValue = "true")
     public boolean transformUnproxyableClasses;
 
+    /**
+     * If set to true, if a private method that is neither an observer nor a producer and is annotated with an interceptor
```
Comment (mkouba) [inline extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcConfig.java:73]: _"..., the build fails if a private..."_?

Comment (geoand) [reply extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcConfig.java]: Fixed

**Review by mkouba — APPROVED** [review]

**Comment by mkouba** [comment]
The tests from the `integration-tests/grpc-plain-text-mutiny` seem to pass locally. Let's wait for the next CI results and then merge.

**Pull request merged**